numpy/core/fromnumeric.py

```diff
@@ -1272,8 +1293,16 @@ def argmin(a, axis=None, out=None):
     >>> np.take_along_axis(x, np.expand_dims(index_array, axis=-1), axis=-1).squeeze(axis=-1)
     array([2, 0])
 
+    Setting `keepdims` to `True`,
+
+    >>> x = np.arange(24).reshape((2, 3, 4))
+    >>> res = np.argmin(x, axis=1, keepdims=True)
+    >>> res.shape
+    (2, 1, 4)
     """
-    return _wrapfunc(a, 'argmin', axis=axis, out=out)
+    if isinstance(a, np.matrix):
+        return _wrapfunc(a, 'argmin', axis=axis, out=out)
```

Comment:
We probably should juse use `keepdims=np._NoValue` and not `None` here. The mechanism used in a few places that `NoValue` translates to just dropping the argument entirely.  (I am not sure if `_wrapfunc` already deals with that without checking.)

`matrix` is important (and probably already buggy with `out`), but we should avoid hardcoding it.  (Since we would like to remove it as soon as SciPy stops relying on it for sparse.)